File: python/bubble_sort.py
# Your Code Here
def bubble_sort(lst, swps=0):
    start=swps
    swaps= swps
    for idx in range(len(lst)-1):
        if lst[idx]>lst[idx+1]:
            holder=lst[idx+1]
            lst[idx+1]=lst[idx]
            lst[idx]=holder
            swaps+=1
    if swaps == start:
        print(f"Final result: {lst}")
        print(f"Swaps: {swaps}")
    else:
        bubble_sort(lst, swaps)
        
sequence = [4, 3, 5, 0, 1]        
bubble_sort(sequence)

File: js/bubbleSort.js
// Your Code Here
const bubbleSort =(arr, swp=0)=>{
    let start= swp;
    let swaps=swp;
    for(let idx=0; idx<arr.length -1; idx++){
        if (arr[idx]>arr[idx+1]){
            holder=arr[idx+1]
            arr[idx+1]=arr[idx]
            arr[idx]=holder
            swaps++
        }
    }
    if (swaps ===start){
        console.log("Final result: " + arr)
        console.log("Swaps: " + swaps)
    }
    else{
        bubbleSort(arr, swaps)
    }
}

var sequence = [4, 3, 5, 0, 1]
bubbleSort(sequence)
